### DISCUSSION TITLE
Running WW3 on Apple Silicon Mac

### DISCUSSION THREAD

**Discussion by joragan** [NOAA-EMC/WW3#1044] [Q&A]
Hi folks,

Has anyone been able to run WW3 on an Apple Silicon-based Mac? I'm working to understand the basics of the tool local before I try and get it running on a remote server and I've been trying to follow the disparate instructions that seem to hint that running on a Mac should work but I've been unable to successfully get WW3 compiling. Before I spend further time I wanted to see if anyone had been able to get the app running or if this was only possible on older intel based Macs.

Cheers!

**Comment by aliabdolali** [comment]
There is a wiki page for MAC
https://github.com/NOAA-EMC/WW3/wiki/FAQs-page#For-MACOSX-with-gfortran

**Comment by joragan** [comment]
Yah, I had found that part of the FAQ before but it seems either that it's out of date with Apple Silicon Macs or I'm having some other issue.

When I get to the following steps:

```
**WW3 Compilation**
cd bin
ln -s comp.Gnu comp
ln -s link.Gnu link
cp switch<user_defined> switch
./w3_make
```

I get the following error output from the make:

```
bin % ./w3_make
 
                *****************************
              ***   WAVEWATCH III make      ***
                *****************************
 
Exporting WWATCH3_ENV=/Users/dk/Dev/WW3/model/bin/wwatch3.env
Main directory    : /Users/dk/Dev/WW3/model
Scratch directory : /Users/dk/Dev/WW3/model/tmp
Save source codes : yes
Save listings     : yes
 
[ERROR] comp file /Users/dk/Dev/WW3/model/bin/comp not found
        Please run /Users/dkDev/WW3/model/bin/w3_setup <main_dir> -c <comp> -s <switch>
```

Looking into the bin directory there was no comp.Gnu or link.Gnu for the ln to target in the first place. Not sure if those are supposed to be created by the w3_setup, but they didn't seem to be created and I don't see any errors from the setup run

```
model % ./bin/w3_setup ./        
 
                *****************************
              ***   WAVEWATCH III setup     ***
                *****************************
 
 
[INFO] local env file wwatch3.env found in /Users/dk/Dev/WW3/model/bin/wwatch3.env
   Setup file /Users/dk/Dev/WW3/model/bin/wwatch3.env found
      Source directory            : /Users/dk/Dev/WW3/model
      Scratch directory           : /Users/dk/Dev/WW3/model/tmp
      Save source code            : yes
      Save listings               : yes
   Update settings ? [y/n] n
 
   Create required model subdirectories
 
Finished setting up WAVEWATCH III
```

To that end I've been trying to better understand what the different 'comp' and 'switch' mean but I'm not seeing an obvious definition set in the documentation. Is there a specific place I'm missing?

Thanks for the help!

**Reply by JessicaMeixner-NOAA** [reply]
The different "comp" files are for different sets of intel or gnu compiler options.  There's several for specific HPC machines. 

For switches, you can find information in section 5.9 "Selecting model options" in the manual.

**Comment by CarstenHansen** [comment]
From the command-line standard output you show it seems to me like you are setting up the _develop_ version, not the _main_ branch (WW3 v 6.07.1 release).

In _develop_  the former 'comp' files (comp.Gnu, link.Gnu etc.) are not present but have been replaced by templates (comp.tmpl, link.tmpl, and also ad3.tmpl). Invoking the setup command like `./bin/w3_setup ./ -c gnu_debug -s NCEP_st4 -t ./tmp` will create the files bin/comp, bin/link and bin/ad3. To keep local modifications you may copy those files to bin/comp.Gnu_debug etc.

I haven't checked if this is described in the _develop_ branch version of the User manual.

**Reply by joragan** [reply]
@CarstenHansen ah, that explains why I didn't see the files. I'd suggest that the wiki be updated to not check out the _develop_ branch (or at least have a note about how to create the files to be link / change to the _main_ branch). I'm unfortunately not fully out of the woods as now the _w3_setup_ call is erroring, but I'm going to experiment/debug a bit more on my end before asking for more help here.

**Side question:**
Before I spend a bunch of extra time on this I think it prudent to make sure my ending expectation is correct. For context, I'm working on a small commercial project that allow a user to present current swell and wave conditions and their future forcast for a geographical area to the users of the product. I'm starting with a focus on the Southern California region. From sites like swell matrix and storm glass I can see from their data sources that they list _WW3_ as one of the main points where they get this data. However in my reading I'm also seeing _GEFS-Wave_ listed and want to make sure that for the type of use case I'm envisioning that trying to get WW3 running is they correct way forward or is there a better tool or usage of _WW3_ or other NOAA output that would produce the type of data that I'd be looking for?

**Reply by aliabdolali** [reply]
Is your need for the past events or forecasted data?

**Reply by joragan** [reply]
@aliabdolali If a user was to select a location (i.e. Cabrillo Beach, south of LA near the Palos Verdes peninsula), I'd be looking to display current swell/waves for that local area, as well as a forecast for swell/waves into the future. A way to check current and expected future conditions to help decide when to get into the water or not. I don't current have an obvious need for past conditional data.

**Reply by aliabdolali** [reply]
I think those data are publicly a available for a few days (current and future). Since your region is coastal, higher resolution data is better for your application than coarse outputs (nwps). The available data are: 
1- Global forecast System GFSv16. Google it and you will find the ftp url. Resolution is 0.25 degrees and deterministic
2- Global Ensemble forecast System GEFSv12. Google it and you will find the ftp url. Resolution is 0.25 degrees. You also can ask @ricampos 
3- Nearshore wave prediction system (nwps). This one is SWAN based but high resolution and only coastal. You can ask @AliS-Noaa 

You can find all you want (i.e. Swell) from those three product or can build your own setup, but in the case of running it yourself, you need boundary condition from a global model like gfs, otherwise you won't get swell right plus you need at least wind forcing to run your setup.

Hope it was helpful
